nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/BulkRecords.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * (C) Copyright 2018 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *       Kevin Leturc <kleturc@nuxeo.com>
+ */
+package org.nuxeo.ecm.core.bulk;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.nuxeo.lib.stream.computation.Record;
+
+/**
+ * Helper class to handle {@link Record} in bulk services.
+ *
+ * @since 10.2
+ */
+public class BulkRecords {
```

Comment:
Helper again, with a DTO you don't need this
Command myCommand = codec.decode(record.getData())
myCommand.getId()
myCommand.getDocIds()
Should be the same for Bulk and Counter object

Reply:
Indeed, we'll done it in https://jira.nuxeo.com/browse/NXP-25301.